조이-안지유/movie/src/pages/SignUp.jsx

```diff
@@ -2,54 +2,66 @@ import React from 'react';
 import styled from "styled-components";
 import SignUpInput from "../components/SignUpInput.jsx";
 import {useForm} from "react-hook-form";
+import {useNavigate} from "react-router-dom";
 
 const SignUp = () => {
   const {
     register,
     handleSubmit,
     watch,
-    setError,
     formState: {errors},
   } = useForm({
     mode: "onChange" // 입력 필드가 변경될 때마다 유효성검사 실행
   });
-  const checkPassword = watch("password");
+  const password = watch("password");
   const onSubmit = (data) => {
-    if (data.password !== data.checkPassword) {
-      setError("checkPassword", {
-        type: "manual",
-        message: "비밀번호가 일치하지 않습니다.",
-      });
-      return;
-    }
     console.log(data);
   }
+
   return (
     <SignUpContainer>
       <Title>회원가입 페이지</Title>
       <SignUpForm onSubmit={handleSubmit(onSubmit)}>
-        <SignUpInput name="name" type="text" placeholder="이름을 입력해주세요" {...register("name", {required: "이름을 입력해주세요!"})}
-                     errors={errors}/>
-        <SignUpInput name="email" type="email"
-                     placeholder="이메일을 입력해주세요" {...register("email", {required: "이메일을 입력해주세요!"})} errors={errors}/>
-        <SignUpInput name="age" type="text" placeholder="나이를 입력해주세요" {...register("age", {
-          required: "나이를 입력해주세요!",
-          min: {value: 19, message: "19세 이상만 이용가능합니다!"},
-          pattern: {value: /^\d+$/, message: "나이는 숫자로 입력해주세요!"}
+        <Input name="name" type="text" placeholder="이름을 입력해주세요" {...register("name", {
```

Comment:
처음에 인풋 컴포넌트(SignUpInput.jsx)를 따로 분리했었는데, register 넘기는 과정에서 자꾸 에러가 나서 분리하지 않고 SignUp.jsx 파일 내에서만 했습니다! 그러니 SignUpInput.jsx 파일은 일단은 무시하고 봐주시면 감사하겠습니다.... ^__^....
컴포넌트 분리하는 건 다시 고민해보겠습니다 .... ㅠ__ㅠ

Reply:
저도 input 컴포넌트를 분리하면 더 좋겠다는 생각은 했는데 넘기는 과정에서 무슨 에러가 났는지 궁금합니당 
넘기면 아예 못받고 있는지 아니면 넘기긴 하는데 그 시점이 register가 없는지..! 아래처럼하면 어떻게 도ㅣ나요
```suggestion
        <Input name="name" type="text" placeholder="이름을 입력해주세요" {...(register && register("name", { requited: "이름을 입력해주세요!" }))}
```

Reply:
그리고 input 컴포넌트가 많을 때 모듈화를 잘 했다고 생각한 블로그를 찾아서 혹시 관심있으면 [여기](https://velog.io/@pengoose_dev/%EB%94%94%EC%9E%90%EC%9D%B8-%ED%8C%A8%ED%84%B4%EC%9D%84-%EC%9D%B4%EC%9A%A9%ED%95%9C-React-Hook-Form%EC%9D%98-%ED%95%9C%EA%B3%84-%EA%B7%B9%EB%B3%B5)도 참고해보세용

Reply:
감사합니다🙇‍♀️🩷🩷🩷

Reply:
register 값을 넘기면 에러메시지로 register가 함수가 아니라고,, 그렇게 떴습니다 🥲 그리고 또 고민이 됐던 게, 코드에서와 같이 스프레드 연산자를 이용해서 register를 넘기는 과정에서 첫 번째 파라미터에는 name을, 두 번째 파라미터에는 option을 추가해서 validate를 수행했는데 이 option이 인풋마다 다 달라서 이 또한 어떻게 넘겨야할지 고민을 많이 했었고,, 아직도 하고 있습니다!,,(진행중)